src/main/java/org/folio/service/titles/TitlesService.java

```diff
@@ -130,16 +160,175 @@ private List<String> getNonPackageLineIds(List<CompositePoLine> compositePoLines
     return compositePoLines.stream().filter(line -> !line.getIsPackage()).map(CompositePoLine::getId).collect(toList());
   }
 
-  public Future<String> updateTitleWithInstance(String titleId, RequestContext locationContext, RequestContext requestContext) {
+  public Future<String> updateTitleWithInstance(String titleId, RequestContext locationContext,
+                                                RequestContext requestContext) {
     return getTitleById(titleId, requestContext)
       .compose(title -> updateTitleWithInstance(title, false, locationContext, requestContext));
   }
 
-  public Future<String> updateTitleWithInstance(Title title, boolean isInstanceMatchingDisabled, RequestContext locationContext, RequestContext requestContext) {
+  public Future<String> updateTitleWithInstance(Title title, boolean isInstanceMatchingDisabled,
+                                                RequestContext locationContext, RequestContext requestContext) {
     return titleInstanceService.getOrCreateInstance(title, isInstanceMatchingDisabled, locationContext)
       .map(title::withInstanceId)
       .compose(entity -> saveTitle(entity, requestContext)
         .map(v -> entity.getInstanceId()));
   }
 
+  /**
+   * Unlinks a title from a package. This involves checking the existence of holdings and items for the title.
+   * If the holdings have references to other titles, the unlinking process skips them.
+   * If the holdings do not have references, it asks for confirmation to delete related pieces, items, and holdings.
+   *
+   * @param titleId        the ID of the title to unlink
+   * @param deleteHolding  flag indicating whether to delete the holding if it has no other references
+   * @param requestContext the request context
+   * @return a Future representing the result of the unlinking operation
+   */
+  public Future<List<String>> unlinkTitleFromPackage(String titleId, String deleteHolding,
+                                                     RequestContext requestContext) {
+    log.debug("Trying to unlink title with id: {} and deleteHolding: {}", titleId, deleteHolding);
+
+    return getTitleById(titleId, requestContext)
+      .compose(title -> purchaseOrderLineService.getOrderLineById(title.getPoLineId(), requestContext)
+        .compose(poLine -> processHoldingItemPieces(poLine, title, deleteHolding, requestContext))
+        .compose(holdings -> deleteTitle(title, holdings, requestContext)));
```

Comment:
I think we don't want to delete the title here, just unlink it. There is a different endpoint (`DELETE /orders/titles/{id}`) to delete titles.

Reply:
If we want to unlink a title without deleting it, is to add a new boolean flag called ‘linked’ to the Title schema. This flag can be used to filter out titles from the Package PO line when linked=false. In this case, we would be able to re-link the title again when needed.
We asked about this in https://folio-org.atlassian.net/browse/MODORDERS-1260